src/Simulation.cpp

```diff
@@ -316,12 +316,31 @@ void Simulation::readXML(XMLfileUnits& xmlconfig) {
 			else if(parallelisationtype == "KDDecomposition") {
 				delete _domainDecomposition;
 				_domainDecomposition = new KDDecomposition(getcutoffRadius(), _domain, _ensemble->getComponents()->size());
-			}
-			else if(parallelisationtype == "GeneralDomainDecomposition") {
-              delete _domainDecomposition;
-              _domainDecomposition = new GeneralDomainDecomposition(getcutoffRadius(), _domain);
-            }
-			else {
+			} else if (parallelisationtype == "GeneralDomainDecomposition") {
+				double skin = 0.;
+				// we need the skin here, so we extract it from the AutoPas container's xml,
+				// because the ParticleContainer needs to be instantiated later. :/
+				xmlconfig.changecurrentnode("..");
+				if (xmlconfig.changecurrentnode("datastructure")) {
+					string datastructuretype;
+					xmlconfig.getNodeValue("@type", datastructuretype);
+					if (datastructuretype == "AutoPas" || datastructuretype == "AutoPasContainer") {
+						xmlconfig.getNodeValue("skin", skin);
+						global_log->info() << "Using skin = " << skin << " for the GeneralDomainDecomposition." << std::endl;
+					} else {
+						global_log->error() << "Using the GeneralDomainDecomposition is only supported when using "
+											   "AutoPas, but AutoPas is not used."
```

Comment:
```suggestion
											   "AutoPas, but the configuration file does not use it."
```
Or something like this to make clear that the problem lies in the config file and not (necessarily) the compilation step.